### PR TITLE
Updated `ouroboros-consensus-diffusion` to `ouroboros-network-0.16.0.0`

### LINKED ISSUE #974: Refine Bootstrap State Machine to ignore local peers when it should
https://github.com/IntersectMBO/ouroboros-network/issues/4815 would allow the Bootstrap State Machine to distinguish remote from local peers, which is sometimes crucial. See that issue for more details on what the GSM code (introduced in https://github.com/IntersectMBO/ouroboros-consensus/pull/808) should do once that information is available.

### PR DESCRIPTION
**Pull request by coot**
Closes #974

### CODE REVIEW

File: ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs [hunk 1]
```diff
@@ -239,9 +246,12 @@ initNodeKernel args@NodeKernelArgs { registry, cfg, tracers
                     gsmMarkerFileView
               , GSM.writeGsmState = \x -> atomically $ do
                   writeTVar varLedgerJudgement $ GSM.gsmStateToLedgerJudgement x
-              , -- In the context of bootstrap peers, it is fine to always
-                -- return 'True' as all peers are trusted during syncing.
-                GSM.isHaaSatisfied            = pure True
+              , GSM.isHaaSatisfied            = do
+                  readTVar varOutboundConnectionsState <&> \case
+                    -- See the upstream Haddocks for the exact conditions under
+                    -- which the diffusion layer is in this state.
```
Comment (amesgen) [inline ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs:252]: For convenience: [upstream Haddocks](https://github.com/IntersectMBO/ouroboros-network/blob/c5314093f33c01f910be59b24f092be8ceeff9b1/ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LocalRootPeers.hs#L9)

**Review by nfrisby — COMMENTED** [review]
LGTM

Reviewed on a call. Esgen is going to add something to the Bootstrap Peers IER.

**Comment by amesgen** [comment]
> Esgen is going to add something to the Bootstrap Peers IER.

[Rendered Bootstrap Peers IER](https://github.com/IntersectMBO/ouroboros-consensus/blob/coot/ouroboros-network-0.16/docs/website/contents/for-developers/BootstrapPeersIER.md)

**Comment by nfrisby** [comment]
@coot wrote on Slack

> I would write a sentence that in the `pre-syncing` state the node will wait for `Honest Availability Assumption`. And maybe expand what it actually means for `chain-sync` clients.

**Review by jasagredo — APPROVED** [review]

**Comment by amesgen** [comment]
FTR: We want to merge this in order to unblock the release train for Node 8.11; we can merge improvements to the docs in a follow-up PR.

**Pull request merged**